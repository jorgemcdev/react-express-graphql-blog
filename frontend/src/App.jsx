import React, { lazy, Suspense } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { Container } from 'semantic-ui-react';

import 'semantic-ui-css/semantic.min.css';
import './App.css';

import { AuthProvider } from './context/auth';

import AuthRoute from './utils/AuthRoute';

import MenuBar from './components/menuBar/MenuBar';

const Home = lazy(() => import('./pages/home/Home'));
const Login = lazy(() => import('./pages/login/Login'));
const Register = lazy(() => import('./pages/register/Register'));
const SinglePost = lazy(() => import('./pages/singlePost/SinglePost'));

const App = () => (
  <AuthProvider>
    <Router>
      <Container>
        <MenuBar />
        <Suspense fallback={<div>Loading</div>}>
          <Route exact path="/" component={Home} />
          <AuthRoute path="/login" component={Login} />
          <AuthRoute path="/register" component={Register} />
          <Route exact path="/posts/:postId" component={SinglePost} />
        </Suspense>
      </Container>
    </Router>
  </AuthProvider>
);

export default App;
