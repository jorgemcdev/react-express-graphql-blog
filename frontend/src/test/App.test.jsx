/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from '../App';

describe('AUTOCOMPLETE: autocomplete cities', () => {
  it('Should Match the Snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should Render Autcomplete', () => {
    const wrapper = shallow(<App />);
    const title = wrapper.find('h1');
    expect(title.length).toEqual(1);
  });
});
