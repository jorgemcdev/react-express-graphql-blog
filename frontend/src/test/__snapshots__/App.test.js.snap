// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AUTOCOMPLETE: autocomplete cities Should Match the Snapshot 1`] = `
Array [
  <h1>
    My App
  </h1>,
  <p>
    Hello React
  </p>,
  <input>
    Click
  </input>,
]
`;
